usr/lib/common/defs.h

```diff
@@ -17,6 +17,20 @@
 #ifndef _DEFS_H
 #define _DEFS_H
 
+#include <openssl/opensslv.h>
+
+#ifndef OPENSSL_VERSION_PREREQ
+    #if defined(OPENSSL_VERSION_MAJOR) && defined(OPENSSL_VERSION_MINOR)
+        #define OPENSSL_VERSION_PREREQ(maj, min)        \
+            ((OPENSSL_VERSION_MAJOR << 16) +        \
+            OPENSSL_VERSION_MINOR >= ((maj) << 16) + (min))
+    #else
+        #define OPENSSL_VERSION_PREREQ(maj, min)        \
+            (OPENSSL_VERSION_NUMBER >= (((maj) << 28) | \
```

Comment:
Well, then check is if OPENSSL_VERSION_NUMBER is larger or equal to the version number built from may and min. So if OPENSSL_VERSION_NUMBER includes a pre-relaese in the 4 lowest bits, then it would still be larger or equal to the version built from may and min, given that the upper bits parts match. 
If you remove (i.e. zeroize) the 4 lower bits, then it would not make a big difference, any pre-release would still match.

Reply:
Well, "remove" was a bad choice of words.  What I meant is "should a pre-release match?"

Reply:
> should a pre-release match?

I would say yes. The whole block is only used for pre OpenSSL 3.0 anyway, because in OpenSSL 3.0 OPENSSL_VERSION_PREREQ is already defined.
So if there ever is a pre-release version for 1.1.1 then I guess I would still be able to compile against it. It will anyway match 1.1.x, because it only checks major and minor. So I guess your question does not really apply :-)